### PR TITLE
feat: make data tables support html

### PR DESCRIPTION
**Pull request by mistercrunch**
### SUMMARY
allowing links and other basic HTML to render in data tables safely. This tackles:
- in SQL Lab
- in the Explore south pane in `Results`
- in the Explore south pane in `Samples`
- in "drill to details" menu picker (removes html tags if any)
- in the table view of Drill to details

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screen Shot 2023-06-12 at 2 23 41 PM
Screen Shot 2023-06-12 at 2 23 26 PM
Screen Shot 2023-06-12 at 5 44 15 PM
Screen Shot 2023-06-12 at 5 48 58 PM

### TESTING INSTRUCTIONS
* create an expression or calculated dim
* render in SQL Lab, samples, result set view

### CODE REVIEW

File: superset-frontend/packages/superset-ui-core/src/utils/html.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { FilterXSS, getDefaultWhiteList } from 'xss';
+
+const xssFilter = new FilterXSS({
+  whiteList: {
```
[the remark this answers is not in the document]
Comment (rusackas) [reply superset-frontend/packages/superset-ui-core/src/utils/html.tsx:5]: Sorry, didn't realize this was legacy code being moved over... it just kinda jumped out at me :)

[the remark this answers is not in the document]
Comment (mistercrunch) [reply superset-frontend/packages/superset-ui-core/src/utils/html.tsx:5]: yeah I wonder if a newer version of the `xss` lib has alternative semantics

**Comment by worker24h** [comment]
image

The feature is very best，I need it too.  

But, I hope to that, **Add config item**  by Web UI ，rather than ** CONCAT In sql  ** 

What are your ideas ？ @mistercrunch

**Review by eschutho — APPROVED** [review]
🎉

**Comment by rusackas** [comment]
> The feature is very best，I need it too.
> 
> But, I hope to that, **Add a config item** by Web UI ，rather than **CONCAT In sql** ，
> 
> What are your ideas ？ @mistercrunch
> 
> image

I think the concat is nice in that it allows you to re-use the columnar data anywhere needed in the link. For example, a Github issue ID is used as the linked text AND a particular spot in the URL. If we were to add a link to the UI, we would need some form of syntax (e.g. handlebars/jinja) that would allow you to construct the link in the correct way... and that can have security-related concerns.

I love the idea of it being in the UI, but I'm having a hard time thinking of a way to do it that's both simple and secure.

**Comment by nytai** [comment]
Hmm, I was just looking into adding this support to the pivot tables too.

**Comment by worker24h** [comment]
> > The feature is very best，I need it too.
> > But, I hope to that, **Add a config item** by Web UI ，rather than **CONCAT In sql** ，
> > What are your ideas ？ @mistercrunch
> > image
> 
> I think the concat is nice in that it allows you to re-use the columnar data anywhere needed in the link. For example, a Github issue ID is used as the linked text AND a particular spot in the URL. If we were to add a link to the UI, we would need some form of syntax (e.g. handlebars/jinja) that would allow you to construct the link in the correct way... and that can have security-related concerns.
> 
> I love the idea of it being in the UI, but I'm having a hard time thinking of a way to do it that's both simple and secure.

YES, I got it, thanks

**Pull request merged**